contracts/src/rewarder/HyperdriveCheckpointSubrewarder.sol

```diff
@@ -0,0 +1,156 @@
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+contract HyperdriveCheckpointSubrewarder is IHyperdriveCheckpointSubrewarder {
+    using SafeERC20 for ERC20;
+
+    /// @notice The checkpoint rewarder's name.
+    string public name;
+
+    /// @notice The checkpoint rewarder's version.
+    string public constant version = VERSION;
+
+    /// @notice The rewarder address that can delegate to this subrewarder.
+    address public immutable rewarder;
+
+    /// @notice The checkpoint rewarder's admin address.
+    address public admin;
+
+    /// @notice The address that is the source for the reward funds.
+    address public source;
+
+    /// @notice The associated registry. This is what will be used to determine
+    ///         which instances should receive checkpoint rewards.
+    IHyperdriveRegistry public registry;
+
+    /// @notice The reward token.
+    IERC20 public rewardToken;
+
+    /// @notice The non-trader reward amount.
+    uint256 public minterRewardAmount;
+
+    /// @notice The trader reward amount.
+    uint256 public traderRewardAmount;
+
+    /// @notice Instantiates the hyperdrive checkpoint rewarder.
+    /// @param _name The checkpoint rewarder's name.
+    /// @param _rewarder The address of the rewarder.
+    /// @param _registry The address of the registry.
+    constructor(
+        string memory _name,
+        address _rewarder,
+        IHyperdriveRegistry _registry
+    ) {
+        admin = msg.sender;
+        rewarder = _rewarder;
+        registry = _registry;
+        name = _name;
+    }
+
+    /// @dev Ensures that the modified function is only called by the admin.
+    modifier onlyAdmin() {
+        if (msg.sender != admin) {
+            revert IHyperdriveCheckpointSubrewarder.Unauthorized();
+        }
+        _;
+    }
+
+    /// @dev Ensures that the modified function is only called by the rewarder.
+    modifier onlyRewarder() {
+        if (msg.sender != rewarder) {
+            revert IHyperdriveCheckpointSubrewarder.Unauthorized();
+        }
+        _;
+    }
+
+    /// @notice Allows the admin to transfer the admin role.
+    /// @param _admin The new admin address.
+    function updateAdmin(address _admin) external onlyAdmin {
+        admin = _admin;
+        emit AdminUpdated(_admin);
+    }
+
+    /// @notice Allows the admin to update the registry.
+    /// @param _registry The new registry.
+    function updateRegistry(IHyperdriveRegistry _registry) external onlyAdmin {
+        registry = _registry;
+        emit RegistryUpdated(_registry);
+    }
+
+    /// @notice Allows the admin to update the reward token.
+    /// @param _rewardToken The new reward token.
+    function updateRewardToken(IERC20 _rewardToken) external onlyAdmin {
+        rewardToken = _rewardToken;
+        emit RewardTokenUpdated(_rewardToken);
+    }
+
+    /// @notice Allows the admin to update the source.
```

Comment:
Suggestion to make it more clear:

    /// @notice Allows the admin to update the source address containing the rewards